### PR TITLE
More tmpfile + misc

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

File: glnx-fdio.c [hunk 1]
```diff
@@ -244,6 +244,27 @@ glnx_open_tmpfile_linkable_at (int dfd,
...
+{
+  if (!glnx_open_tmpfile_linkable_at (AT_FDCWD, "/var/tmp", flags, out_tmpf, error))
+    return FALSE;
+  if (out_tmpf->path)
+    {
+      (void) unlinkat (out_tmpf->src_dfd, out_tmpf->path, 0);
+      g_clear_pointer (&out_tmpf->path, g_free);
+    }
+  /* This special value means "anonymous" */
+  out_tmpf->src_dfd = -3;
```
Comment (jlebon) [inline glnx-fdio.c]: Can we just add an `anonymous` bool to the struct for this?

Comment (cgwalters) [reply glnx-fdio.c]: Sure, done :arrow_down:

**Comment by jlebon** [comment]
👍

**Comment by cgwalters** [comment]
Merged, thanks!

**Pull request closed without merging**